### PR TITLE
[DOP-11906] Improve Clickhouse documentation

### PR DESCRIPTION
**Pull request by dolfinus**
## Change Summary

* Improved Clickhouse's documentation about reading, writing and executing statements. Previously it was just crosslinks to DBReader, DBWriter and connection own methods docstrings. Now there are more Clickhouse-specific examples and recommendations
* Added new section describing type compatibility between Clickhouse and Spark, including some recommendations for reading/writing unsupported column types.
* Added new section describing which actions should be performed to connect to Clickhouse.

See https://onetl--211.org.readthedocs.build/en/211/connection/db_connection/clickhouse/index.html

If new documentation layout will be approved, we can drop generic examples from ``JDBC.fetch()``, ``JDBC.execute()``, ``JDBC.sql()``, ``JDBC.JDBCOptions`` and so on, and replace them with DB-specific documentation and examples.

## Checklist

* [X] Commit message and PR title is comprehensive
* [X] Keep the change as small as possible
* [ ] Unit and integration tests for the changes exist
* [ ] Tests pass on CI and coverage does not decrease
* [X] Documentation reflects the changes where applicable
* [X] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/onetl/blob/develop/CONTRIBUTING.rst) for details.)
* [X] My PR is ready to review.

### CODE REVIEW

**Review by maxim-lixakov — APPROVED** [review]

File: docs/connection/db_connection/clickhouse/prerequisites.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Required grants
+~~~~~~~~~~~~~~~
+
+Ask your Clickhouse cluster administrator to set following grants for a user,
+used for creating a connection:
+
+.. tabs::
+
+    .. code-tab:: sql Read + write
```
Comment (dmitry-pedchenko) [inline docs/connection/db_connection/clickhouse/prerequisites.rst:47]:
```suggestion
    .. code-tab:: sql Read + Write
```

File: docs/connection/db_connection/clickhouse/read.rst [hunk 2]
```diff
@@ -1,18 +1,74 @@
...
 
-.. automethod:: Clickhouse.sql
+* ✅︎ ``columns``
+* ✅︎ ``where``
+* ✅︎ ``hwm``, supported strategies:
+* * ✅︎ :ref:`snapshot-strategy`
+* * ✅︎ :ref:`incremental-strategy`
+* * ✅︎ :ref:`snapshot-batch-strategy`
+* * ✅︎ :ref:`incremental-batch-strategy`
+* ❌ ``hint`` (not supported by Clickhouse)
```
Comment (dmitry-pedchenko) [inline docs/connection/db_connection/clickhouse/read.rst:23]:
```suggestion
* ❌ ``hint`` (does not supported by Clickhouse)
```

File: docs/connection/db_connection/clickhouse/sql.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    clickhouse = Clickhouse(...)
+    df = clickhouse.sql(
+        """
+        SELECT
+            id,
+            key,
+            CAST(value AS String) value,
+            updated_at
+        WHERE
```
Comment (dmitry-pedchenko) [inline docs/connection/db_connection/clickhouse/sql.rst:42]: а тут FROM не надо? в клике такой синтаксис?

**Review by dmitry-pedchenko — APPROVED** [review]

File: docs/connection/db_connection/clickhouse/sql.rst (hunk 3 above)
Comment (dolfinus) [reply docs/connection/db_connection/clickhouse/sql.rst:44]: Исправил

**Pull request merged**